frontend/packages/kubevirt-plugin/src/hooks/use-permissions-card-permissions.ts

```diff
@@ -0,0 +1,132 @@
+import { useAccessReview2 } from '@console/internal/components/utils';
+import { TemplateModel } from '@console/internal/models';
+import { K8sVerb } from '@console/internal/module/k8s/types';
+import { NetworkAttachmentDefinitionModel } from '@console/network-attachment-definition-plugin';
+import { useFlag } from '@console/shared';
+import { KUBEVIRT_OS_IMAGES_NS, NAMESPACE_OPENSHIFT, OPENSHIFT_OS_IMAGES_NS } from '../constants';
+import { FLAG_KUBEVIRT_CDI } from '../flags/const';
+
+export type UsePermissionsCardPermissions = {
+  capabilitiesData: { [key: string]: boolean };
+  permissionsLoading: boolean;
+  numAllowedCapabilities: number;
+  numNotAllowedCapabilities: number;
+};
+
+export const usePermissionsCardPermissions = (): UsePermissionsCardPermissions => {
+  const cdiInstalled = useFlag(FLAG_KUBEVIRT_CDI);
+
+  const [canReadOpenshiftNs, canReadOpenshiftNsLoading] = useAccessReview2({
+    namespace: NAMESPACE_OPENSHIFT,
+    verb: 'get' as K8sVerb,
+    resource: TemplateModel.plural,
+  });
+
+  const [canWriteToOpenshiftNs, canWriteToOpenshiftNsLoading] = useAccessReview2({
+    namespace: NAMESPACE_OPENSHIFT,
+    verb: 'create' as K8sVerb,
+    resource: TemplateModel.plural,
+  });
+
+  const [canReadOpenshiftOsImgNs, canReadOpenshiftOsImgNsLoading] = useAccessReview2({
+    namespace: OPENSHIFT_OS_IMAGES_NS,
+    verb: 'get' as K8sVerb,
+    resource: TemplateModel.plural,
+  });
+
+  const [canReadKvOsImgNs, canReadKvOsImgNsLoading] = useAccessReview2({
+    namespace: KUBEVIRT_OS_IMAGES_NS,
+    verb: 'get' as K8sVerb,
+    resource: TemplateModel.plural,
+  });
+
+  const [canWriteToKvOsImgNs, canWriteToKvOsImgNsLoading] = useAccessReview2({
+    namespace: KUBEVIRT_OS_IMAGES_NS,
+    verb: 'create' as K8sVerb,
+    resource: TemplateModel.plural,
+  });
+
+  const [canReadNads, canReadNadsLoading] = useAccessReview2({
+    verb: 'get' as K8sVerb,
+    resource: NetworkAttachmentDefinitionModel.plural,
+  });
+
+  // establish base permissions for all tasks
+  const canReadOsImagesNs = canReadKvOsImgNs && canReadOpenshiftOsImgNs;
+  const canReadOsImagesNsLoading = canReadKvOsImgNsLoading && canReadOpenshiftOsImgNsLoading;
+
+  const hasBasePermissions = canReadOpenshiftNs && canReadOsImagesNs;
+  const hasBasePermissionsLoading = canReadOpenshiftNsLoading && canReadOsImagesNsLoading;
+
+  // access public templates task
+  const canAccessPublicTemplates = hasBasePermissions;
+  const canAccessPublicTemplatesLoading = hasBasePermissionsLoading;
+
+  // access public boot sources task
+  const canAccessPublicBootSources = hasBasePermissions;
```

Comment:
canAccessPublicBootSources === canAccessPublicTemplates ?
both of them are just hasBasePermissions , consider using a different more generic name and declare once? as they are the same

Reply:
I changed the way I tracked these, so this comment and the loading one should be irrelevant now.